pkg/controllers/provisioning/scheduling/nodeclaim.go

```diff
@@ -173,95 +171,92 @@ type filterResults struct {
 	// fitsAndOffering indicates if a single instance type had enough resources and was a required offering
 	fitsAndOffering          bool
 	minValuesIncompatibleErr error
-	requests                 v1.ResourceList
+
+	// We capture requirements so that we can know what the requirements were when evaluating instance type compatibility
+	requirements scheduling.Requirements
+	// We capture podRequests here since when a pod can't schedule due to requests, it's because the pod
+	// was on its own on the simulated Node and exceeded the available resources for any instance type for this NodePool
+	podRequests v1.ResourceList
+	// We capture daemonRequests since this contributes to the resources that are required to schedule to this NodePool
+	daemonRequests v1.ResourceList
 }
 
-// FailureReason returns a presentable string explaining why all instance types were filtered out
-//
 //nolint:gocyclo
-func (r filterResults) FailureReason() string {
-	if len(r.remaining) > 0 {
-		return ""
-	}
-
+func (e InstanceTypeFilterError) Error() string {
 	// minValues is specified in the requirements and is not met
-	if r.minValuesIncompatibleErr != nil {
-		return r.minValuesIncompatibleErr.Error()
+	if e.minValuesIncompatibleErr != nil {
+		return fmt.Sprintf("%s, requirements=%s, resources=%s", e.minValuesIncompatibleErr.Error(), e.requirements, resources.String(resources.Merge(e.daemonRequests, e.podRequests)))
 	}
-
 	// no instance type met any of the three criteria, meaning each criteria was enough to completely prevent
 	// this pod from scheduling
-	if !r.requirementsMet && !r.fits && !r.hasOffering {
-		return "no instance type met the scheduling requirements or had enough resources or had a required offering"
+	if !e.requirementsMet && !e.fits && !e.hasOffering {
+		return fmt.Sprintf("no instance type met the scheduling requirements or had enough resources or had a required offering, requirements=%s, resources=%s", e.requirements, resources.String(resources.Merge(e.daemonRequests, e.podRequests)))
```

Comment:
Does the improvement require that `e.requirements, resources.String(resources.Merge(e.daemonRequests, e.podRequests)` be called in every branch as opposed to once at the top of `.Error()`?